ktor-features/ktor-auth-jwt/src/io/ktor/auth/jwt/JWTAuth.kt

```diff
@@ -0,0 +1,99 @@
+package io.ktor.auth.jwt
+
+import com.auth0.jwk.*
+import com.auth0.jwt.*
+import com.auth0.jwt.algorithms.*
+import com.auth0.jwt.exceptions.*
+import com.auth0.jwt.impl.*
+import com.auth0.jwt.interfaces.*
+import io.ktor.application.*
+import io.ktor.auth.*
+import io.ktor.http.*
+import io.ktor.response.*
+import java.security.interfaces.*
+import java.util.*
+
+private val JWTAuthKey: Any = "JWTAuth"
+
+data class JWTCredential(val payload: Payload) : Credential
```

Comment:
does `Payload` provide `hashCode` and `equals`?

Reply:
No. PayloadImpl is a plain class, no `equals` or `hashCode.`

Reply:
`toString()`? why these classes are data classes?

Reply:
I removed the data class modifier in the latest commit.